src/data/text_parser.h

```diff
@@ -120,34 +119,24 @@ inline bool TextParserBase<IndexType, DType>::FillData(
   const char *head = reinterpret_cast<char *>(chunk.dptr);
 #pragma omp parallel num_threads(nthread)
   {
-    try {
-      // threadid
-      int tid = omp_get_thread_num();
-      size_t nstep = (chunk.size + nthread - 1) / nthread;
-      size_t sbegin = std::min(tid * nstep, chunk.size);
-      size_t send = std::min((tid + 1) * nstep, chunk.size);
-      const char *pbegin = BackFindEndLine(head + sbegin,
-                                           head);
-      const char *pend;
-      if (tid + 1 == nthread) {
-        pend = head + send;
-      } else {
-        pend = BackFindEndLine(head + send,
-                               head);
-      }
-      ParseBlock(pbegin, pend, &(*data)[tid]);
-    } catch (dmlc::Error& ex) {
-      {
-        std::lock_guard<std::mutex> lock(mutex_exception_);
-        if (!parser_exception_) {
-          parser_exception_ = std::current_exception();
-        }
-      }
+  omp_exc_.Run([=] {
```

Comment:
I'm a bit wary about the default value capture here. For example, is it cheap to copy `chunk`?

I think we should stick to capture by reference `[&]`, since there is no danger of the lambda outliving the captured objects.

Reply:
good point. you are right it should be by reference.  i fixed it at other places in mxnet code but missed it here.